### PR TITLE
659 compilation for sync settings

### LINKED ISSUE #659: Compilation for sync settings
This branch does not compile: https://github.com/openmsupply/open-msupply/pulls

Need to propagate mutation/query changes to syncSettings.

Also as part of this would be great to do error mapping in structured way.

Ended up with more changes then I wanted to add to one PR, so broken down into:

- [ ] https://github.com/openmsupply/open-msupply/pull/671
   * Want to include url and flatten down MappedError, move variants of SyncErrorV5 to the same level as MappedError.  
   * Reformat thiserror derives slightly, they don't need to always provide 'source' as this can be traversed (like what anyhow::Error does)
   * Add specific error for SyncError rather then anyhow::Error (easier to map underlying SynApiError and better formatting for thiserror
- [ ] https://github.com/openmsupply/open-msupply/pull/672
- [ ] https://github.com/openmsupply/open-msupply/pull/673
- [ ] https://github.com/openmsupply/open-msupply/pull/677
- [ ] https://github.com/openmsupply/open-msupply/pull/679

### LINKED ISSUE #669: 'Unable to sync' error while synchronisation still in progress (setting up new client Android)
With mSupply Server running on a laptop and open-msupplyclient (phone hotspot) setting up a Lenovo Tablet

After entering sync settings the ui waits for a bit then returns an error 'Unable to sync. Please check your settings'.
However on the server side we can see that synchronisation is still in progress.

This is with Approximately 50,000 sync records via a cellphone hotspot (but local connection)

After waiting for the sync to finish records appear to be available.

**Expected Behaviour**
- Ideal: Sync Progress Bar
- Ok: Some kind of ui saying 'still syncing' instead of error message

### LINKED ISSUE #670: 'Server Error' on login for recently setup android device
Related to #669

After setting up and syncing a new android device, after  sync appears to have finished on the server side, login fails with an generic 'Server Error' message on the client.

There are no requests received on the central server to validate the password.

Since there were a lot of records in this sync site (~50,000) it could be that the site records are still being 'integrated'?

After waiting a few minutes more login starts working.

**Expected behaviour**
- More informative error message or logins should succeed during integration (if indeed that is the issue)

### PR DESCRIPTION
**Pull request by andreievg**
closes #659

also should close:

closes #669 
closes #670 

This is head of stacked PRs with some minor changes.

### Tests

Tests taken from: https://github.com/openmsupply/open-msupply/pull/640

- [ ] With fresh database, start server and f/e
- [ ] Should see sync progress
- [ ] Enter credentials and press initialise
- [ ] Should see progress
- [ ] Should be navigated to login page on completion
- [ ] Stop server, remove database and `cargo run`
- [ ] Add something to throw an error in 4d `syncV5API_queuedRecordsGet`
- [ ] Try initialise (should still error)
- [ ] Retry button should become visible and enabled while retrying
- [ ] Try retrying (prepare initial should not fire again (can see by the speed of that step)
- [ ] Try refresh page when error happens (or while initialising) state should persist
- [ ] Try clearing database again and entering settings in yaml file (still with error is in 4d method)
- [ ] Open front end and you should see the correct state
- [ ] Remove error in 4D
- [ ] Press retry again (sync should successfully complete and you should be navigated to login)

Other tests:

After initialisation in sync settings page (need admin permissions in store)

- [ ] Try wrong url, username and password (should result in readable message
- [ ] Try changing hardware id for the site in central (should result in readable message)
- [ ] Try changing to a different site (should result in message about UUID being changed)
- [ ] Try stoping central and serving on another port, when changing url to new port should be sucesfull.
- [ ] Change interval, save and refresh page, interval should persist.
- [ ] Clicking on error icon should show a notification (this is in sync progress in initialisation, verification error in initialisation above initialisation button, and in sync settings page)

### CODE REVIEW

**Comment by mark-prins** [comment]
Are you sure about #674? that one seems off topic

File: client/packages/common/src/ui/components/inputs/TextInput/PasswordTextInput.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  (props, ref) => {
+    // if the helper text is a space then the height of the component doesn't change
+    // when the helper text is shown / removed
+    const defaultWarning = props.fixedHeight ? ' ' : '';
+    const [showPassword, setShowPassword] = useState(false);
+    const [warning, setWarning] = useState(defaultWarning);
+    const t = useTranslation();
+    const visibilityInputButton = (
+      <IconButton
+        aria-label="toggle password visibility"
```
Comment (mark-prins) [inline client/packages/common/src/ui/components/inputs/TextInput/PasswordTextInput.tsx:21]: Can I suggest an addition?
```
       disabled={props.disabled}
```

when on the initialisation screen, after clicking initialise, the password input is disabled but you can still click the magic eye..

**Review by mark-prins — APPROVED** [review]
I confess, that I gave up looking through all the files after about 10%. there's just too many!

Have run the site, executed manual testing, run the b/e and f/e tests. 
Had a look at the issues linked and checked that they are probably fixed. Have not created a new android build and tested that yet. I suggest merging then building android and retesting those issues.

**Comment by andreievg** [comment]
> I confess, that I gave up looking through all the files after about 10%. there's just too many!

I had to update from develop at some point, was going to leave it till last, but there was a request to introduce new password component, so once I've update this branch from develop the diff becomes huge (since base branch wasn't updated from develop). Sorry, but yeah it only had 200 changes originally.

File: client/packages/common/src/ui/components/inputs/TextInput/PasswordTextInput.tsx (hunk 1 above)
Comment (andreievg) [reply client/packages/common/src/ui/components/inputs/TextInput/PasswordTextInput.tsx:21]: 🙏 de1926edebbe71c78929649d74cb972c986ae03b and I tested

**Pull request merged**